servant-client-core/src/Servant/Client/Core/Internal/Request.hs

```diff
@@ -65,14 +66,15 @@ type Request = RequestF Builder.Builder
 newtype RequestBody = RequestBodyLBS LBS.ByteString
   deriving (Eq, Ord, Read, Show, Typeable)
 
-data Response = Response
+data GenResponse a = Response
   { responseStatusCode  :: Status
-  , responseBody        :: LBS.ByteString
   , responseHeaders     :: Seq.Seq Header
   , responseHttpVersion :: HttpVersion
-  } deriving (Eq, Show, Generic, Typeable)
+  , responseBody        :: a
+  } deriving (Eq, Show, Generic, Typeable, Functor, Foldable, Traversable)
 
-data StreamingResponse = StreamingResponse { runStreamingResponse :: forall a. ((Status, Seq.Seq Header, HttpVersion, IO BS.ByteString) -> IO a) -> IO a }
+type Response = GenResponse LBS.ByteString
+newtype StreamingResponse = StreamingResponse { runStreamingResponse :: forall a. (GenResponse (IO BS.ByteString) -> IO a) -> IO a }
```

Comment:
Shouldn't this, for symmetry, be something like 
```
newtype StreamingBody = StreamingBody (forall a. ((IO ByteString) -> a) -> a)
type StreamingResponse = GenResponse StreamingBody
```

(not sure if those details are right, but what I'm referring to is having the continuation *within* `GenResponse` rather than containing it)

Reply:
It won't work. see `performStreamingRequest`